src/java/htsjdk/samtools/SAMUtils.java

```diff
@@ -558,10 +566,31 @@ public static void makeReadUnmapped(final SAMRecord rec) {
         rec.setMappingQuality(SAMRecord.NO_MAPPING_QUALITY);
         rec.setInferredInsertSize(0);
         rec.setNotPrimaryAlignmentFlag(false);
+        rec.setSupplementaryAlignmentFlag(false);
         rec.setProperPairFlag(false);
         rec.setReadUnmappedFlag(true);
     }
 
+    /**
+     * Strip mapping information from a SAMRecord, but preserve it in the 'O' tags if it isn't already set.
+     */
+    public static void makeReadUnmappedWithOriginalTags(final SAMRecord rec) {
+        if (!hasOriginalMappingInformation(rec)) {
+            rec.setAttribute(SAMTag.OP.name(), rec.getAlignmentStart());
```

Comment:
while this is already merged, we need to store the original Seqence index or name....right?